ts/Series/Treegraph/TreegraphPoint.ts

```diff
@@ -0,0 +1,205 @@
+import { Palette } from '../../Core/Color/Palettes';
+const { merge, addEvent, pick, fireEvent } = U;
+const {
+    seriesTypes: {
+        treemap: {
+            prototype: {
+                pointClass: TreemapPoint
+            }
+        }
+    }
+} = SeriesRegistry;
+
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+interface ShapeArgs {
+    x: number;
+    y: number;
+    height: number;
+    width: number;
+}
+
+class TreegraphPoint extends TreemapPoint {
+    public options: TreegraphPointOptions = void 0 as any;
+    public isLink = false;
+    public collapseButton?: SVGElement;
+    public series: TreegraphSeries = void 0 as any;
+    public collapsed?: boolean;
+    public node: TreegraphNode.Node = void 0 as any;
+    public level?: number;
+    public linkToParent?: TreegraphLink;
+
+    draw(): void {
+        super.draw.apply(this, arguments);
+        this.renderCollapseButton();
+    }
+
+    renderCollapseButton(): void {
+        const point = this,
+            series = point.series,
+            parentGroup = point.graphic && point.graphic.parentGroup,
+            levelOptions =
+                (series.mapOptionsToLevel as any)[point.node.level || 0] || {},
+            btnOptions = merge(
+                series.options.collapseButton,
+                levelOptions.collapseButton,
+                point.series.options.collapseButton
+            ) as CollapseButtonOptions,
+            { width, height, shape, style } = btnOptions,
+            padding = 2,
+            chart = this.series.chart;
+        if (!point.shapeArgs) {
+            return;
+        }
+
+        this.options.collapseButton = btnOptions;
+        if (!point.collapseButton) {
+            if (!point.node.children.length || !btnOptions.enabled) {
+                return;
+            }
+            let { x, y } = this.getCollapseBtnPosition(btnOptions);
+            point.collapseButton = chart.renderer
+                .label(
+                    point.collapsed ? '+' : '-',
+                    x,
+                    y,
+                    shape
+                )
+                .attr({
+                    height: height - 2 * padding,
+                    width: width - 2 * padding,
+                    padding: padding,
+                    fill: Palette.neutralColor20,
+                    rotation: chart.inverted ? 90 : 0,
+                    rotationOriginX: width / 2,
+                    rotationOriginY: height / 2,
+                    stroke: Palette.neutralColor80,
+                    'stroke-width': 1,
+                    'text-align': 'center',
+                    align: 'center',
+                    zIndex: 1
+                })
+                .addClass('highcharts-tracker')
+                .addClass('highcharts-collapse-button')
+                .removeClass('highcharts-no-tooltip')
+                .css(style || {})
+                .add(parentGroup);
+
+            (point.collapseButton.element as any).point = point;
```

Comment:
It's somewhat bad practice to put this reference to a DOM element. Can we refer it somewhere else?

Reply:
Isn't it the go-to way of making the custom renderer object have the same tooltip as a point?